zingolib/src/lightclient/send.rs

```diff
@@ -35,44 +35,59 @@ pub mod send_with_proposal {
             proposal: &Proposal<zip317::FeeRule, OutputRef>,
             sending_account: zip32::AccountId,
         ) -> Result<NonEmpty<TxId>, SendError> {
-            let mut wallet = self.wallet.write().await;
-            let calculated_txids = wallet
-                .calculate_transactions(proposal, sending_account)
-                .await
-                .map_err(SendError::CalculateSendError)?;
-            self.latest_proposal = None;
-
-            Ok(wallet
-                .transmit_transactions(self.server_uri(), calculated_txids)
-                .await?)
+            match self.server_uri() {
+                None => Err(SendError::UnreachableIndexer),
```

Comment:
i dont think this is necessary and bloats the errors tree. we should be able to construct and store a tx offline. it is transmission that requires a URI and you have already added the error to transmission errors which is passed back through SendError via the `from` impl